fili-core/src/main/java/com/yahoo/bard/webservice/data/config/table/PhysicalTableDefinition.java

```diff
@@ -1,93 +1,144 @@
+     * Get the set of physical tables required to build the current physical table.
+     *
+     * @return a set of physical table names
+     */
+    public abstract Set<TableName> getDependentTableNames();
+
+    /**
+     * Given a set of metric names and a data source metadata service, build the corresponding physical table.
      *
-     * @param name  The table name
-     * @param grain  The zoneless time grain
-     * @param dimensions  The dimension configurations
+     * @param dictionaries  Dictionary containing dimension dictionary and physical table dictionary
+     * @param metadataService  Service containing column available interval information
      *
-     * @deprecated The time zone of a physical table should be set explicitly rather than rely on defaulting to UTC
+     * @return optional of the corresponding type of physical table, empty optional if could not be build.
      */
-    @Deprecated
-    public PhysicalTableDefinition(
-            TableName name,
-            ZonelessTimeGrain grain,
-            Iterable<? extends DimensionConfig> dimensions
-    ) {
-        this(name, grain.buildZonedTimeGrain(DateTimeZone.UTC), dimensions);
-    }
+    public abstract Optional<PhysicalTable> build(
+            ResourceDictionaries dictionaries,
+            DataSourceMetadataService metadataService
+    );
 
     public TableName getName() {
         return name;
     }
 
-    public ZonedTimeGrain getGrain() {
-        return grain;
+    public ZonedTimeGrain getTimeGrain() {
+        return timeGrain;
     }
 
-    public ImmutableSet<? extends DimensionConfig> getDimensions() {
-        return dimensions;
+    public Set<FieldName> getMetricNames() {
+        return metricNames;
+    }
+
+    public Set<? extends DimensionConfig> getDimensionConfigs() {
+        return dimensionConfigs;
     }
 
     public Map<String, String> getLogicalToPhysicalNames() {
         return logicalToPhysicalNames;
     }
+
+    /**
+     * Builds the dimension logical name to physical name mapping from dimension configs.
+     *
+     * @param dimensionConfigs  Dimension config containing both logical and physical names
+     *
+     * @return the dimension logical name to physical name mapping
+     */
+    protected Map<String, String> buildLogicalToPhysicalNames(Set<? extends DimensionConfig> dimensionConfigs) {
+        return dimensionConfigs.stream()
+                .collect(
+                        Collectors.toMap(
+                                DimensionConfig::getApiName,
+                                config -> {
+                                    String physicalName = config.getPhysicalName();
+                                    if (physicalName == null) {
+                                        throw new RuntimeException("No physical name found for dimension: "
+                                                + config.getApiName());
+                                    }
```

Comment:
Perhaps not a warning, but log _something_ so that it's not surprising that we're filling in the blanks. Info or Debug levels would be fine by me, as long as the method indicates it's going to do this.